### PR TITLE
Feature/package www

### PR DESCRIPTION
**Pull request by kalikaneko**

### CODE REVIEW

File: service/pixelated/resources/root_resource.py [hunk 1]
```diff
@@ -40,10 +40,10 @@
 
 
 class RootResource(BaseResource):
-    def __init__(self, services_factory):
+    def __init__(self, services_factory, static_folder=None):
```
Comment (tuliocasagrande) [inline service/pixelated/resources/root_resource.py]: Are we passing custom static folders?

Comment (kalikaneko) [inline service/pixelated/resources/root_resource.py]: I don't understand the question precisely.
They are "custom" in the sense that I need to control where they are (ie, inside a virtualenv or on the top-level folder from a froze bundle).
I find this solution more straightforward that the _get_static_folder() below, which seems a bit hacky.

Comment (tuliocasagrande) [inline service/pixelated/resources/root_resource.py]: I agree the `_get_static_folder()` looks a bit hacky.
Maybe we could remove it and add the default at `arguments.py`, like: `RootResource(services_factory, args.static_folder)`

**Comment by kalikaneko** [comment]
Adding a bit more to the discussion:

1. the option of getting the static files distributed within the main (python) pixelated-user-agent. I dislike this idea because forces a contributor to have all the nodejs + ruby packaging machinery. I think that, for the point of view of a python backend developer, it's simpler to do "pip install -U pixelated-www" and get the latest and greatest of the frontend.

2. the issue of synchronization between pixelated-user-agent and pixelated-www: I believe this could be solved by a runtime check that enforces they have matching versions. If pixelated-user-agent moves to the same schema (0.1.x.timestamp), then it's easy to force the matching timestamp. 

3. helping the above, the main pixelated-user-agent package can declare an extra (setuptools extras, like pixelated['www'] that specifies the compiled-ready  dependency. it can force the version- timestamp in there too I believe.

4. the extra python www package must be seen as a convenient help to development, not something needed in the tool chain. for instance, debian package for pixelated might well ignore the -www phython one and ship everything for final distribution.

5. I think "releases" (or better, snapshots) could be automated at some point, to publish wheels to a certain authoritative site (somewhere at pixelated, like I did here: https://downloads.leap.se/libs/pixelated/). Nightlies would be great. At some point (weekely?) a pypi release would be awesome too.

No rush to get this included, for now I'll use this strategy to ship pixelated in our bundles and will publish the wheels to internal leap urls.

**Comment by deniscostadsc** [comment]
I'm not sure this is the best place to start this discission. But I'll do it any way :bowtie:.

Do you believe that the best way have different things in different place we could separate frontend in a second repository? @bwagnerr once told me that this would be good if this don't get things harder for external contributors.

What do you think?

:heart:

**Comment by kalikaneko** [comment]
hmm I don't see a direct benefit on splitting it into a different repo, but neither do I see a downside from it. from our side, having the static data and the python code as separate packages as this PR proposes is enough.

**Comment by bwagnerr** [comment]
I think it would be good, especially considering they are totally different languages with different setups, I would prefer to have them in different repositories. I still don't know how to make it easy for contributors if we split.

**Comment by mnandri** [comment]
Well, we could combine both ideas.
1. break it into two repos
2. run the user-agent in such a way that we either provide where the ui is served, so user-agent is pure REST API.  Or where the ui static files are located so that it is served from the user-agent as well, the same as it is currently (pip packaged separately or not).

The benefits are:
- clear separations: you can improve on the backend without necessarily deep diving on the frontend and vice versa...  this could make things much easier for contributors... also, clear contract tests can be defined/enforced (over the API side and vice-versa on the frontend).
- allowing for ease of various clients development/integration: Android, iOS and who knows windows :)
- work/stories could be well compartimentalized, to backend and/or frontend tasks, if desired.
- scaling needs can be catered/customized better.
- it's just more natural: e.g. why would ./go setupjs be in the service folder?

Disadvantages:
- maintaining two repos
- may need some work on the CI/CD pipelines 
- we need to publish and document the user-agent API specs
- additional work on smoothing setup for contributors (for the best though)
- there may be some security issues... that needs to be assessed and addressed.

The question boils down to: is the repo large enough or not synchronised enough for it to warrant the extra work that it will require to split it.

**Comment by mnandri** [comment]
For the record.

We have decided that:
- Find ways to make the pixelated_www package as is specified by kalyi above, from our existing setup.py (in service): corresponding issues  https://github.com/pixelated/pixelated-user-agent/issues/729 and https://github.com/pixelated/pixelated-user-agent/issues/730
- Find ways to move the go command to the root folder in the pixelated repo: corresponding issue https://github.com/pixelated/pixelated-user-agent/issues/731

**Comment by mnandri** [comment]
In the mean time, I'm merging this.

**Pull request merged**